### PR TITLE
Unit tests for date time methods

### PR DESCRIPTION
**Pull request by Salipa-Gurung**
## Description
Added unit tests for methods related to formatting date and time.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added

### CODE REVIEW

File: tests/unit/App.spec.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    })
+
+    describe('method: formatDateAndTime', () => {
+      it('should format a valid date', () => {
+        expect(wrapper.vm.formatDateAndTime('19960823', '093801')).toBe('Aug 23, 1996, 9:38 AM')
+      })
+      it('should return undefined for invalid date or invalid time', () => {
+        expect(wrapper.vm.formatDateAndTime()).toBe(undefined)
+        expect(wrapper.vm.formatDateAndTime('1996082', '093801')).toBe(undefined)
+        expect(wrapper.vm.formatDateAndTime('1996082', '09380')).toBe(undefined)
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:103]: they look similar. comments?
```suggestion
        expect(wrapper.vm.formatDateAndTime('1996082', '093801')).toBe(undefined)
        expect(wrapper.vm.formatDateAndTime('1996082', '09380')).toBe(undefined)
```

File: tests/unit/App.spec.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    describe('method: formatDateAndTime', () => {
+      it('should format a valid date', () => {
+        expect(wrapper.vm.formatDateAndTime('19960823', '093801')).toBe('Aug 23, 1996, 9:38 AM')
+      })
+      it('should return undefined for invalid date or invalid time', () => {
+        expect(wrapper.vm.formatDateAndTime()).toBe(undefined)
+        // Testing for invalid date argument => length of date less than 8
+        expect(wrapper.vm.formatDateAndTime('1996082', '093801')).toBe(undefined)
+        // Testing for invalid time argument => length of time less than 6
+        expect(wrapper.vm.formatDateAndTime('19960823', '09380')).toBe(undefined)
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:107]: maybe they deserve their own `it`? and there you can have list of invalid data to test

File: tests/unit/App.spec.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          date: "1996082",
+          time: "09380",
+        },
+      ];
+      it("should format a valid date", () => {
+        expect(wrapper.vm.formatDateAndTime("19960823", "093801")).toBe(
+          "Aug 23, 1996, 9:38 AM"
+        );
+      });
+      it.each(invalidDateTimeCombinations)(
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:130]: no need for extra variable if not reused
```suggestion
      it.each([
        {
          date: undefined,
          time: undefined,
        },
        {
          date: "1996082",
          time: "093801",
        },
        {
          date: "19960823",
          time: "09380",
        },
        {
          date: "1996082",
          time: "09380",
        },
      ])(
```

File: tests/unit/App.spec.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        `should return undefined for invalid date or invalid time`,
+        ({ date, time }) => {
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:132]: create dynamic test descriptions. would be easier to see the test cases covered
```suggestion
        `should return undefined for %s`,
        ({ testcase, date, time }) => {
```

**Comment by koebel** [comment]
The current implementation of formatDate() and formatTime() also take the bool to define the output format. I think this distinction has so far not been reflected in the tests. We could also refactor the code and just pass the output format into the function... that might be the better approach anyway.

**Comment by koebel** [comment]
> The current implementation of formatDate() and formatTime() also take the bool to define the output format. I think this distinction has so far not been reflected in the tests. We could also refactor the code and just pass the output format into the function... that might be the better approach anyway.

I've actually moved these two functions into the new helper class and did the suggested refactoring regarding output format, see https://github.com/owncloud/web-app-dicom-viewer/pull/16 (still in draft status)

File: src/App.vue [hunk 5]
```diff
@@ -195,6 +195,7 @@ export default defineComponent({
       viewport: null,
       viewportCameraParallelScale: 1,
       dicomUrl: null,
+      dicomImageData:null,
```
Comment (saw-jan) [inline src/App.vue:198]: has usage?

File: tests/unit/App.spec.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+      });
+    });
+
+
+    describe("method: formatDate", () => {
+      it("should format a valid date", () => {
+        expect(wrapper.vm.formatDate("19010101")).toBe("1/1/1901");
+      });
+      it.each([undefined, "1901010"])(
+        "should return undefined for invalid date",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:86]:
```suggestion
        "should return undefined for invalid date: %s",
```

File: tests/unit/App.spec.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+          date: "19960823",
+          time: "093801",
+          expected: "Aug 23, 1996, 9:38 AM",
+          description: "valid date and valid time",
+        },
+        {
+          date: undefined,
+          time: undefined,
+          expected: undefined,
+          description: "invalid date or invalid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:105]:
```suggestion
          description: "invalid date and invalid time",
```

File: tests/unit/App.spec.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        },
+        {
+          date: "1996082",
+          time: "093801",
+          expected: undefined,
+          description: "invalid date or invalid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:111]:
```suggestion
          description: "invalid date and valid time",
```

File: tests/unit/App.spec.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        },
+        {
+          date: "19960823",
+          time: "09380",
+          expected: undefined,
+          description: "invalid date or invalid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:117]:
```suggestion
          description: "valid date and invalid time",
```

File: tests/unit/App.spec.ts [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+        },
+        {
+          date: "1996082",
+          time: "09380",
+          expected: undefined,
+          description: "invalid date or invalid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:123]:
```suggestion
          description: "invalid date and time",
```

File: tests/unit/App.spec.ts [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+      );
+    });
+
+    describe("method: formatDateAndTime", () => {
+      it.each([
+        {
+          date: "19960823",
+          time: "093801",
+          expected: "Aug 23, 1996, 9:38 AM",
+          description: "valid date and valid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:98]:
```suggestion
          description: "valid date and time",
```

File: tests/unit/App.spec.ts [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+          date: "19960823",
+          time: "09380",
+          expected: undefined,
+          description: "valid date and invalid time",
+        },
+        {
+          date: "1996082",
+          time: "09380",
+          expected: undefined,
+          description: "invalid date and invalid time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:116]:
```suggestion
          description: "invalid date and time",
```

File: tests/unit/App.spec.ts [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+        },
+        {
+          date: undefined,
+          time: undefined,
+          expected: undefined,
+          description: "undefined date and undefined time",
```
Comment (saw-jan) [inline tests/unit/App.spec.ts:122]:
```suggestion
          description: "undefined date and time",
```

Comment (saw-jan) [reply tests/unit/App.spec.ts:122]: Please, check the changes requested

**Review by saw-jan — APPROVED** [review]
:+1:

**Review by koebel — APPROVED** [review]
LGTM

If you still think that it's a bit confusing that we have two functions called "formatDate()" in the app (the one here in app.vue and the other one in the helper function which formats the date to a different output format), then rename one of them to whatever you think is less confusing.

**Pull request merged**